src/UserAgreement/Quiz.jsx

```diff
@@ -0,0 +1,192 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Question from './Question';
+import QuizRight from './QuizRight';
+import './Quiz.less';
+
+const getQuestionSectionId = idx => `Q${idx}`;
+
+const makeDefaultState = (questions) => {
+  const notDone = [];
+  questions.forEach(
+    (item, i) => {
+      notDone.push(i);
+    },
+  );
+  return {
+    notDone,
+    rights: [],
+    wrongs: [],
+    currentSection: getQuestionSectionId(1),
+  };
+};
+
+/**
+ * Little quiz component - roperites: questionList, title, onSubmit
+ */
+class Quiz extends Component {
+  static propTypes = {
+    questionList: PropTypes.arrayOf(
+      PropTypes.shape(
+        {
+          name: PropTypes.string,
+          question: PropTypes.string,
+          options: PropTypes.arrayOf(PropTypes.string),
+          answer: PropTypes.number,
+          hint: PropTypes.string,
+        },
+      ),
+    ),
+    title: PropTypes.string.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    questionList: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.qs = {};
+    this.qsDom = {};
+    this.state = makeDefaultState(this.props.questionList);
+    this.resetState = this.resetState.bind(this);
+    this.scrolledByClick = false;
+  }
+
+  componentDidMount() {
+    this.menuHeight = this.menu.offsetHeight;
+    this.qs_list.addEventListener('scroll',
+      () => this.handleScroll(this.qs_list), this.menuHeight);
+    const currentQ = this.qsDom[getQuestionSectionId(Object.values(this.qsDom).length)];
+    this.latestOffsetTop = currentQ.offsetTop;
+  }
+
+  componentWillUnmount() {
+    this.qs_list.removeEventListener('scroll',
+      () => this.handleScroll(this.qs_list));
+  }
+
+  setCurrentSection(sectionIdx) {
+    this.setState({ currentSection: getQuestionSectionId(sectionIdx + 1) });
+  }
+
+  resetState() {
+    this.setState(makeDefaultState(this.props.questionList));
+  }
+
+  answerQuestion(questionId, isCorrect) {
+    const notDone = this.state.notDone.filter(item => item !== questionId);
+    let [rights] = this.state.rights;
```

Comment:
is this syntax correct?  I think it can just be `let rights...`

Reply:
that is the requirement of our Codacy
https://eslint.org/docs/rules/prefer-destructuring